io-threads: move variables to a more memory friendly structure

mykaul opened the pull request:
Accumulating all variables that have to do with a single priority
together makes accessing them (and reading the code) easier.

Made some function static as well, no functional changes.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
io-threads: move variables to a more memory friendly structure

Moving all variables that have to do with a single priority
together makes accessing them (and reading the code) easier.

Made some function static as well, no functional changes.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

amarts commented on xlators/performance/io-threads/src/io-threads.c:
```diff
@@ -111,32 +112,28 @@ __iot_dequeue(iot_conf_t *conf, int *pri)
         *pri = i;
+        conf->queue_size--;
+        return stub;
         break;
```
does this break give a warning saying a 'dead code' in coverity?

mykaul replied:
I can check, but just running clang-scan on the code gives me:

Dead assignment | 539 |  
Dead increment | 2 |  
Dead initialization | 9 |  
Dead nested assignment | 7 |  

So we have more items to fix. (not to mention that we always initialize structs to zero, which many times is not needed - I don't think clang-scan counts those properly!)

mykaul replied:
Removed.

amarts reviewed (commented):
LGTM

itisravi commented on xlators/performance/io-threads/src/io-threads.c:
```diff
@@ -111,32 +112,28 @@ __iot_dequeue(iot_conf_t *conf, int *pri)
+        fop_data->queue_sizes--;
         *pri = i;
+        conf->queue_size--;
+        return stub;
```
I think it is better to retain the ` *pri = -1;` assignment in the beginning and move the `return stub` from here to the end of this function at line 128.

mykaul replied:
For readability reasons? I don't mind, just unsure it makes it more readable.

itisravi replied:
yes,  otherwise, a break after the return looks odd.

mykaul replied:
I can remove the break then - wasn't sure if Coverity or whatnot would not complain. I prefer not to initialize for -1 for no reason.

mohit84 replied:
We don't need to set *pri = -1 that is already set by iot_worker.

mohit84 replied:
I think earlier code was much clean, we don't need to put break after return stub.

Commit:
io-threads: fix review comments

- Do not assign pri to -1 if not needed
- Remove 'break' after 'return' statement
- Remove a call to 'THIS' and use 'conf->this' instead

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mohit84 commented:
/run regression
